mlrun/datastore/base.py

```diff
@@ -139,7 +144,7 @@ def as_df(self, url, subpath, columns=None, df_module=None, format="", **kwargs)
 
         fs = self.get_filesystem()
         if fs:
-            return reader(fs.open(url), **kwargs)
+            return reader(url, **kwargs)
```

Comment:
Are you sure this will also work if the path is `v3io` for example ? 
Even if pandas is fsspec aware - I'm not sure it will know how to initialize the v3iofs instance (which we're doing above)
Also, note that the `df_module` you're using, doesn't have to be pandas, can be provided from outside, I'm assuming it's used for spark df
Lastly, if it is working like this, the `get_filesystem` and the `if` are not needed no ?

Reply:
The `if get_filesystem` part is still needed to retain the existing semantics (there is different handling based on whether there's a scheme to the url or not. I do think that the call should be `self.get_filesystem(silent=False)` though, now that I look at the implementation.

Reply:
The test tests writing to v3io, so it does work. It's needed because you can't call `open` on a directory.

Reply:
So I looked into it, and the if and silent=True thing is actually there so that v3io can fall back on simple HTTP, which I guess would only work for a single file.

Reply:
Adding propagation of storage options.

Reply:
Turns out this is only supported in pandas >= 1.2